admin_guide/diagnostics_tool.adoc

```diff
@@ -157,36 +168,139 @@ A failure from this check indicates that a significant amount of space in etcd
 expiration can be found in
 xref:../install_config/redeploying_certificates.adoc#install-config-redeploying-certificates[Redeploying Certificates].
+====
+
+[[admin-guide-health-checks-via-ansible-playbook]]
+=== Running Health Checks via ansible-playbook
+
+To run the *openshift-ansible* health checks using the `ansible-playbook`
+command, specify your cluster's inventory file and run the *_health.yml_*
+playbook:
+
+----
+# ansible-playbook -i <inventory_file> \
+    /usr/share/ansible/openshift-ansible/playbooks/common/openshift-checks/health.yml
+----
+
+To set variables in the command line, include the `-e` flag with any desired
+variables in `key=value key=value` format. For example:
+
+----
+# ansible-playbook -i <inventory_file> \
+    /usr/share/ansible/openshift-ansible/playbooks/common/openshift-checks/health.yml \
+    -e openshift_check_logging_index_timeout_seconds=30
+    -e etcd_max_image_data_size_bytes=40000000000
+----
+
+To disable specific checks, include the variable `openshift_disable_check` with
+a comma-delimited list of check names in your inventory file before running the
+playbook. For example:
+
+----
+openshift_disable_check=etcd_traffic,etcd_volume
+----
+
+Alternatively, set any checks you want to disable as environment variables with `-e openshift_disable_check` when running the `ansible-playbook` command.
+
+[[admin-guide-health-checks-via-docker-cli]]
+=== Running Health Checks via Docker CLI
+
+It is possible a playbook may require dependencies that are not installed
+locally on the host running the `ansible-playbook` command. You can avoid this
+issue by running the *openshift-ansible* health checks directly from the
+ifdef::openshift-enterprise[]
+*ose-ansible*
+endif::[]
+ifdef::openshift-origin[]
+*origin-ansible*
+endif::[]
+image via the Docker CLI.
+
+To do so, specify your cluster's inventory file and the *_health.yml_* playbook
+when running the following `docker run` command as a non-root user that has
+privileges to run containers:
+
+----
+# docker run -u `id -u` \ <1>
+    -v $HOME/.ssh/id_rsa:/opt/app-root/src/.ssh/id_rsa:Z,ro \ <2>
+    -v /etc/ansible/hosts:/tmp/inventory:ro \ <3>
+    -e INVENTORY_FILE=/tmp/inventory \
+    -e PLAYBOOK_FILE=playbooks/byo/openshift-checks/health.yml \ <4>
+    -e OPTS="-v -e openshift_check_logging_index_timeout_seconds=30 -e etcd_max_image_data_size_bytes=40000000000" \ <5>
+ifdef::openshift-enterprise[]
+    openshift3/ose-ansible
+endif::[]
+ifdef::openshift-origin[]
+    openshift/origin-ansible
+endif::[]
+----
+<1> These options make the container run with the same UID as the current user,
+which is required for permissions so that the SSH key can be read inside the
+container (SSH private keys are expected to be readable only by their owner).
+<2> Mount SSH keys as a volume under *_/opt/app-root/src/.ssh_* under normal usage
```

Comment:
The :Z requires some explanation too. As the [repo docs](https://github.com/openshift/openshift-ansible/blob/master/README_CONTAINER_IMAGE.md#usage) explain:

> Note that the ssh key is mounted with the :Z flag: this is also required so that the container can read the ssh key from its restricted SELinux context; this means that your original ssh key file will be re-labeled to something like system_u:object_r:container_file_t:s0:c113,c247. For more details about :Z please check the docker-run(1) man page. Please keep this in mind when providing these volume mount specifications because this could have unexpected consequences: for example, if you mount (and therefore re-label) your whole $HOME/.ssh directory you will block sshd from accessing your keys. This is a reason why you might want to work on a separate copy of the ssh key, so that the original file's labels remain untouched.

ssh is exceedingly picky about what it accepts as keys, and having the wrong SELinux label on the file (which is virtually inevitable at least on the initial run) causes it to silently skip the key and blithely fail to connect without further explanation. It's a pretty infuriating user experience. This is also why it's so important to have the container user matching the user ID on the keyfile.

The mention of mounting a .ssh directory leads to yet more discussion; I don't like it but this probably has to be pretty verbose because this is what everyone will get hung up on adapting to their specific usage. You might want to mount a whole .ssh directory for various reasons, among them because you want to use an ssh config to match keys with hosts or tweak other connection parameters, or because you want to provide a `known_hosts` file and have ssh validate host keys (which is disabled by default config and can be re-enabled by setting the envvar `-e ANSIBLE_HOST_KEY_CHECKING=True` on the cmdline or a few more complicated ways).

Finally we should probably mention providing a vault password. Probably in a future update because frankly I know nothing about it yet.